flake.nix

```diff
@@ -0,0 +1,254 @@
+{
+  description = "Emacs versions for CI";
+
+  inputs = {
+    nixpkgs.url = "nixpkgs/nixpkgs-unstable";
+    flake-utils.url = "github:numtide/flake-utils";
+
+    flake-compat = { url = "github:edolstra/flake-compat"; flake = false; };
+    "emacs-23.4" = { url = "https://ftp.gnu.org/gnu/emacs/emacs-23.4.tar.gz"; flake = false; };
+    "emacs-24.1" = { url = "https://ftp.gnu.org/gnu/emacs/emacs-24.1.tar.gz"; flake = false; };
+    "emacs-24.2" = { url = "https://ftp.gnu.org/gnu/emacs/emacs-24.2.tar.gz"; flake = false; };
+    "emacs-24.3" = { url = "https://ftp.gnu.org/gnu/emacs/emacs-24.3.tar.gz"; flake = false; };
+    "emacs-24.4" = { url = "https://ftp.gnu.org/gnu/emacs/emacs-24.4.tar.gz"; flake = false; };
+    "emacs-24.5" = { url = "https://ftp.gnu.org/gnu/emacs/emacs-24.5.tar.gz"; flake = false; };
+    "emacs-25.1" = { url = "https://ftp.gnu.org/gnu/emacs/emacs-25.1.tar.gz"; flake = false; };
+    "emacs-25.2" = { url = "https://ftp.gnu.org/gnu/emacs/emacs-25.2.tar.gz"; flake = false; };
+    "emacs-25.3" = { url = "https://ftp.gnu.org/gnu/emacs/emacs-25.3.tar.gz"; flake = false; };
+    "emacs-26.1" = { url = "https://ftp.gnu.org/gnu/emacs/emacs-26.1.tar.gz"; flake = false; };
+    "emacs-26.2" = { url = "https://ftp.gnu.org/gnu/emacs/emacs-26.2.tar.gz"; flake = false; };
+    "emacs-26.3" = { url = "https://ftp.gnu.org/gnu/emacs/emacs-26.3.tar.gz"; flake = false; };
+    "emacs-27.1" = { url = "https://ftp.gnu.org/gnu/emacs/emacs-27.1.tar.gz"; flake = false; };
+    "emacs-27.2" = { url = "https://ftp.gnu.org/gnu/emacs/emacs-27.2.tar.gz"; flake = false; };
+    "emacs-28.1" = { url = "https://ftp.gnu.org/gnu/emacs/emacs-28.1.tar.gz"; flake = false; };
+    "emacs-28.2" = { url = "https://ftp.gnu.org/gnu/emacs/emacs-28.2.tar.gz"; flake = false; };
+    emacs-snapshot = { url = "github:emacs-mirror/emacs"; flake = false; };
+    emacs-release-snapshot = { url = "github:emacs-mirror/emacs?ref=emacs-29"; flake = false; };
+    macos-unexec-patches = { url = "https://github.com/emacs-mirror/emacs/commit/888ffd960c06d56a409a7ff15b1d930d25c56089.patch"; flake = false; };
+    latest-package-keyring = { url = "github:emacs-mirror/emacs?path=/etc/package-keyring.gpg"; flake = false; };
+  };
+
+  outputs = { self, nixpkgs, flake-utils, ... }@inputs:
+    flake-utils.lib.eachDefaultSystem (system:
+      rec {
+        overlays.default = self: super: (
```

Comment:
Inside `eachDefaultSystem`/`eachSystem` of `flake-utils`, this will produce per-system overlays, which doesn't make sense. Hence overlay definitions should be outside of the per-system definitions. Nevertheless, I am not sure if this overlay is necessary, as anyone would use this overlay with the pinned nixpkgs to prevent rebuilds.

Reply:
Thanks, yeah, I'll probably also switch things around to avoid the overlay (and maybe remove flake-utils), particularly since the available emacsen vary between platforms. For example, I just noticed that Emacs 26.3 doesn't build on my aarch64-darwin personal machine.

Reply:
You can use `nixpkgs.lib.genAttrs` to get rid of flake-utils, as shown in [this example](https://github.com/nix-systems/nix-systems). You don't need `systems` input here, as you only rebuild against a fixed set of systems.